### ISSUE TITLE
Bad URLs for Sub Projects

### ISSUE THREAD

**Issue by bishopandco** [ericpaulbishop/redmine_git_hosting#12]
Given:
project with an identifier of parent_project and a sub project of sub_project.

When a project is created, the url is stored in the repository as repositories/sub_project

A new repo is created at: 
/home/shared/repositories/sub_project.git

However, another repo is created at /home/shared/repositories/parent_project/sub_project.git

Hope this makes sense.

**Comment by bishopandco** [comment]
This also messes with the clone url
git@server:sub_project.git
should read: 
git@server:parent_project/sub_project.git

**Comment by ericpaulbishop** [comment]
I was able to reproduce and fix the issue with display of subproject urls.  However, I couldn't reproduce the error you report where a second repo is created for a subproject.

**Comment by that0n3guy** [comment]
Eric,  I figured out how to duplicate the issue.  Here is how you do it:

Note: I am using CP (chiliproject) 1.2 mysql
- set the plugin settings so that "Automatically Initialize Git Repositories For New Projects" is enabled.
- create a project called "test" (make sure repository module is checked).
  - this works fine and createds the repo "test.git"
- create another project called "test-sub" (make sure the repository module is checked).
  -  this will create a repo at "test-sub.git" NOT at "test/test-sub.git"
- go to the settings for test-sub project (http://projects.yoursite.com/projects/test-sub/settings).  Switch the "Git Smart HTTP:" to disabled.
  - This will create another repo at "test/test-sub.git".  I now have 2 repos, one at "test-sub.git" and one at "test/test-sub.git"
  - as per this issue: https://github.com/ericpaulbishop/redmine_git_hosting/issues/16 ,  Anything added to the repo at "test/test-sub.git won't show up on the repository page (http://projects.yoursite.com/projects/test-sub/repository) 

So in summary:
- any sub-project created with the repository module checked AND with the "Automatically Initialize Git Repositories For New Projects" enabled. will give you a repo in the main repo directory.
- any sub-project that has the settings for that projects repo changed will get a new repo at "parentproj/subproj.git"
- any project that has the repository module turned on later, then settings changed will add a repo like above.
- the repo's at "parentproj/subproj.git" are not used in the repository view.

**Comment by ericpaulbishop** [comment]
I can confirm that I was able to reproduce the error based on your bug report.  This bug should be fixed now.

**Comment by cocinerox** [comment]
Hi there, sub-sub-projects' (and >=3rd level projects') repositories aren't displayed in the repo tab, because in the 'repositories' mysql table the 'url' and 'root_url' fields contain ".../subproj/subsubproj.git" instead of the valid ".../parentproj/subproj/subsubproj.git" value. See 'lib/git_hosting/patches/projects_controller_patch.rb' line 19. Cheers
